File: test/fixtures/node-axios/post_form.js
import axios from 'axios';
import FormData from 'form-data';

const form = new FormData();
form.append('username', 'davidwalsh');
form.append('password', 'something');

const response = await axios.post(
  'http://localhost:28139/post-to-me.php',
  form,
  {
    headers: {
      ...form.getHeaders()
    }
  }
);

File: test/fixtures/kotlin-ktor/post_form.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.post("http://localhost:28139/post-to-me.php") {
        setBody(
    MultiPartFormDataContent(
        formData {
                append("username", "davidwalsh")
                append("password", "something")
            }
    )
)
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
